kivy/graphics/vertex_instructions_line.pxi

```diff
@@ -1680,43 +1710,41 @@ cdef class SmoothLine(Line):
             tindices[17] = vindex + 7
             tindices = tindices + 18
 
-        if self._close and self._close_mode == 'straight-line':
-            vindex = vcount - 4
-            i0 = vindex
-            i1 = vindex + 1
-            i2 = vindex + 2
-            i3 = vindex + 3
-            i4 = 0
-            i5 = 1
-            i6 = 2
-            i7 = 3
-            tindices[0] = i0
-            tindices[1] = i2
-            tindices[2] = i6
-            tindices[3] = i0
-            tindices[4] = i6
-            tindices[5] = i4
-            tindices[6] = i1
-            tindices[7] = i0
-            tindices[8] = i4
-            tindices[9] = i1
-            tindices[10] = i4
-            tindices[11] = i5
-            tindices[12] = i3
-            tindices[13] = i1
-            tindices[14] = i5
-            tindices[15] = i3
-            tindices[16] = i5
-            tindices[17] = i7
-            tindices = tindices + 18
-
         # print('tindices', <long>tindices, <long>indices, (<long>tindices - <long>indices) / sizeof(unsigned short))
 
-
         self.batch.set_data(vertices, <int>vcount, indices, <int>icount)
 
         free(vertices)
         free(indices)
+    
+    cdef list _remove_close_points(self, list p):
+        cdef int index = 0
+        cdef double x1, y1, x2, y2
+
+        while index < len(p) - 2:
+            x1, y1 = p[index], p[index + 1]
+            x2, y2 = p[index + 2], p[index + 3]
+            if abs(x2 - x1) < 0.1 and abs(y2 - y1) < 0.1:
```

Comment:
Same here for the `0.1` (maybe we can add an argument), so that may be configurable in future.